ui/js/actions/content.js

```diff
@@ -342,16 +351,28 @@ export function doPurchaseUri(uri, purchaseModalName) {
     const costInfo = selectCostInfoForUri(state, { uri });
     const { cost } = costInfo;
 
-    // the file is free or we have partially downloaded it
-    if (cost === 0 || (fileInfo && fileInfo.download_directory)) {
-      dispatch(doLoadVideo(uri));
+    if (cost > balance) {
+      dispatch(doOpenModal(modals.INSUFFICIENT_CREDITS));
       return Promise.resolve();
     }
 
-    if (cost > balance) {
-      dispatch(doOpenModal(modals.INSUFFICIENT_CREDITS));
+    if (
+      cost == 0 ||
+      !lbry.getClientSetting(settings.INSTANT_PURCHASE_ENABLED)
+    ) {
+      attemptPlay(cost);
     } else {
-      dispatch(doOpenModal(purchaseModalName));
+      const instantPurchaseMax = lbry.getClientSetting(
+        settings.INSTANT_PURCHASE_MAX
+      );
+      if (instantPurchaseMax.currency == "LBC") {
+        attemptPlay(cost, instantPurchaseMax.amount);
+      } else {
+        // Need to convert currency of instant purchase maximum before trying to play
+        lbryio.getExchangeRates().then(({ lbc_usd }) => {
```

Comment:
Should this be it's own conversion promise? Wonder if there is a clean way to do this without a promise.

Reply:
It seems like there's some kind of caching going on in `lbryio.getExchangeRates()`, but even then, there's no guarantee that the exchange rates were refreshed recently. We could have it automatically update into the cache at intervals; then we could change `lbryio.getExchangeRates()` to be synchronous and directly return a result from the cache (and/or add a function like `lbryio.convert()` that does the same). Otherwise there's no guarantee that we won't have to go out to the server for this call, so there needs to be a promise or callback.